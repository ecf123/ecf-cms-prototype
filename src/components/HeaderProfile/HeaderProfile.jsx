import React from "react";
import "./HeaderProfile.scss";

const HeaderProfile = ({ children, name, role, toggle }) => {
  React.Children.toArray(children);
  let profile = children[0];
  let arrow = children[1];

  return (
    <div className="header-profile" data-testid="header-profile">
      <div className="header-profile__image">{profile}</div>
      <div className="header-profile__text">
        <h2 className="header-profile__name">{name}</h2>
        <p className="header-profile__role">{role}</p>
      </div>
      <div className="header-profile__arrow" onClick={toggle}>
        {arrow}
      </div>
    </div>
  );
};

export default HeaderProfile;
