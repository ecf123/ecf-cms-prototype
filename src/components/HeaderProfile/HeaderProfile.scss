@use "../../styles/variables" as *;

.header-profile {
  height: 4.25rem;
  width: 14.625rem;
  padding-left: 0.625rem;
  padding-top: 0.563rem;
  padding-bottom: 0.563rem;
  display: flex;
  align-items: center;
  border-radius: 4.375rem;
  background-color: $color-white;

  &__image {
    height: 3.125rem;
    width: 3.125rem;
    margin-right: 0.5rem;
    border-radius: 50%;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  &__text {
    height: fit-content;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
  &__name {
    height: 1.063rem;
    min-width: fit-content;
    margin: 0;
    font-size: $font-size-extra-small;
    font-weight: $font-weight-regular;
    text-align: left;
    color: $color-blue-navy;
    font-family: $font-secondary;
  }
  &__role {
    height: 0.75rem;
    min-width: fit-content;
    margin: 0;
    font-size: $font-size-smallest;
    font-weight: 400;
    text-align: left;
    color: $color-grey-light;
    font-family: $font-secondary;
  }
  &__arrow {
    height: 0.41375rem;
    width: 0.6875rem;
    margin-left: 2.781rem;
    display: flex;
    justify-content: center;
    align-items: center;
    border: none;
    background: none;
  }
}
