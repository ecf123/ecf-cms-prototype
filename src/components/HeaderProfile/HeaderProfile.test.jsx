import { render, screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";
import HeaderProfile from "./HeaderProfile";

it("Should render the HeaderProfile", () => {
  const { container } = customRender(
    <HeaderProfile>
      <p>This is an svg image</p>
      <p>This is an svg image</p>
    </HeaderProfile>
  );
  expect(container).toMatchSnapshot();
});

it("Should render the svg in the Header Profile", () => {
  render(
    <HeaderProfile>
      <p>This is an svg image</p>
      <p>This is an svg image</p>
    </HeaderProfile>
  );
  const svgs = screen.getAllByText("This is an svg image");
  svgs.forEach((svg) => {
    expect(svg).toBeInTheDocument();
  });
});

it("Should display the Name and Role in the HeaderProfile", () => {
  render(
    <HeaderProfile name="Brett Zieme" role="Admin">
      <p>This is an svg image</p>
      <p>This is an svg image</p>
    </HeaderProfile>
  );
  const name = screen.getByText("Brett Zieme");
  const role = screen.getByText("Admin");

  expect(name).toBeInTheDocument();
  expect(role).toBeInTheDocument();
});
