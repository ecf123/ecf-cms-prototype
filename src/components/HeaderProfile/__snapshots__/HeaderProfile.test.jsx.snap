// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the HeaderProfile 1`] = `
<div>
  <div
    class="header-profile"
    data-testid="header-profile"
  >
    <div
      class="header-profile__image"
    >
      <p>
        This is an svg image
      </p>
    </div>
    <div
      class="header-profile__text"
    >
      <h2
        class="header-profile__name"
      />
      <p
        class="header-profile__role"
      />
    </div>
    <div
      class="header-profile__arrow"
    >
      <p>
        This is an svg image
      </p>
    </div>
  </div>
</div>
`;
